sdk/java/src/ai/picovoice/picovoice/Picovoice.java

```diff
@@ -72,50 +74,60 @@ public Picovoice(
             String rhinoModelPath,
             String contextPath,
             float rhinoSensitivity,
+            boolean requireEndpoint,
             PicovoiceInferenceCallback inferenceCallback) throws PicovoiceException {
         try {
             porcupine = new Porcupine.Builder()
+                    .setAccessKey(accessKey)
                     .setLibraryPath(porcupineLibraryPath)
                     .setModelPath(porcupineModelPath)
                     .setSensitivity(porcupineSensitivity)
                     .setKeywordPath(keywordPath)
                     .build();
 
-            if (!porcupine.getVersion().startsWith("1.9.")) {
+            if (!porcupine.getVersion().startsWith("2.0.0")) {
                 final String message = String.format(
-                        "Expected Porcupine library with version '1.9.x' but received %s",
+                        "Expected Porcupine library with version '2.0.x' but received %s",
                         porcupine.getVersion());
                 throw new PicovoiceException(message);
             }
 
             this.wakeWordCallback = wakeWordCallback;
 
             rhino = new Rhino.Builder()
+                    .setAccessKey(accessKey)
                     .setLibraryPath(rhinoLibraryPath)
                     .setModelPath(rhinoModelPath)
                     .setContextPath(contextPath)
                     .setSensitivity(rhinoSensitivity)
+                    .setRequireEndpoint(requireEndpoint)
                     .build();
 
-            if (!rhino.getVersion().startsWith("1.6.")) {
+            if (!rhino.getVersion().startsWith("2.0.0")) {
```

Comment:
We need a check for if ppn/rhn frame length and sample rates are the same. We also should have the callbacks as required arguments (i.e. cannot be null). I can see situations where a customer gives accesskey/ppn/rhn files and then is confused that nothing is happening, so we should encourage them towards adding those callbacks.

Reply:
I see your point but since we are checking the ppn and rhn versions, I think checking ppn/rhn frame length and sample rates are not necessary as we know that those versions are compatible.

Reply:
regardless I added new check points for frame length and sample rates as well as callbacks